bids-validator/validators/tsv/tsv.js

```diff
@@ -344,69 +343,56 @@ const TSV = (file, contents, fileList, callback) => {
-        (val, i) => sampleIdColumnValues.indexOf(val) !== i
-      )
-
-      for (let r = 0; r < rows.length-1; r++) {
-        for (let l = 0; l < rows.length-1; l++) {
-          if (l == r) {}
-          else {
-            if (sampleIdColumnValues[r] == sampleIdColumnValues[l]) {
-              if (participantIdColumnValues[r] != participantIdColumnValues[l]){
-                doesSampleIdHaveDuplicates = false
-              }
-              else {
-                doesSampleIdHaveDuplicates = true
-              }
-            }
-          }
-        }
-      }
-      if(doesSampleIdHaveDuplicates == true) {
-        issues.push(
-          new Issue({
-            file: file,
-            evidence: sampleIdColumnValues,
-            reason: 'Each sample from a same subject MUST be described by one and only one row.',
-            line: 1,
-            code: 220,
-          })
+      // check if a sample from same subject is described by one and only one row
+      let samplePartIdsSet = new Set()
+      for (let r = 0; r < rows.length - 1; r++) {
+        let uniqueString = sampleIdColumnValues[r].concat(
+          participantIdColumnValues[r],
         )
+        // check if SampleId Have Duplicate
+        if (samplePartIdsSet.has(uniqueString)) {
+          issues.push(
+            new Issue({
+              file: file,
+              evidence: sampleIdColumnValues,
+              reason:
+                'Each sample from a same subject MUST be described by one and only one row.',
+              line: 1,
+              code: 220,
+            }),
+          )
+          break
+        } else samplePartIdsSet.add(uniqueString)
       }
-      else {}
-      
     }
-  
 
     // check if any incorrect patterns in sample_type column
     for (let c = 1; c < rows.length; c++) {
       const row = rows[c]
-      if ((row[sampleTypeColumn] != 'cell line') &&
-          (row[sampleTypeColumn] != 'in vitro differentiated cells') &&
-          (row[sampleTypeColumn] != 'primary cell') &&
-          (row[sampleTypeColumn] != 'cell-free sample') &&
-          (row[sampleTypeColumn] != 'cloning host') &&
-          (row[sampleTypeColumn] != 'tissue') &&
-          (row[sampleTypeColumn] != 'whole organisms') &&
-          (row[sampleTypeColumn] != 'organoid') &&
-          (row[sampleTypeColumn] != 'technical sample')) {
-          issues.push(
+      const validSampleTypes = [
+        'cell line',
+        'in vitro differentiated cells',
+        'primary cell',
+        'cell-free sample',
+        'cloning host',
+        'tissue',
+        'whole organisms',
+        'organoid',
+        'technical sample',
+      ]
+      if (!validSampleTypes.includes(row[sampleTypeColumn])) {
+        issues.push(
           new Issue({
             file: file,
             evidence: row[sampleTypeColumn],
-            reason: 'sample_type can\'t be any value.',
+            reason: "sample_type can't be any value.",
             line: c,
```

Comment:
```suggestion
            line: c + 1,
```
The error message works as expected, however I think it needs the `+ 1` otherwise  I got the previous line in the error message.

Reply:
Instead of c + 1, I changed the initial value of c from 1 to 0 in the for loop.

Reply:
I'm not sure if it works because it now seems to check the first line (line 0 with headers) and I get this error:
```
        1: [ERR] sample_type MUST consist of one of the following values: cell line, in vitro differentiated cells, primary cell, cell-free sample, cloning host, tissue, whole organisms, organoid or technical sample. (code: 219 - SAMPLE_TYPE_VALUE)
                ./samples.tsv
                        Evidence: sample_type
```
And the previous error persist, if I put something else than "tissue", I get the error but at wrong line of the files (the line before the error).

Reply:
Do you want the header line to be line 0 or line 1? For now, if we use `line: c`, then line 0 will be the header line.

Reply:
My understanding so far is that the header line is 0 **in the loop**.
We want to skip this line in the check, so start the loop at line c=1 (first line that is not header).

However, because the header line **in the file** is 1, we need the `c + 1` in the error message so it refers to the correct line `in the file`.

Does that makes sense?

Reply:
Yes, makes sense to me. I will update this right now.